### ISSUE TITLE
Volare nix package

### ISSUE THREAD

**Issue by RK4444** [efabless/volare#90]
### Description

I have some issues with importing volare to a nix derivation. I am able to import the volare package but I am not able to execute the volare command. The problem is, that the path since there is no default location in nix is very hard to find (I can't find it right now). So I thought a good idea could be creating a nix package containing volare, since there is no at the moment (atleast i can't find it on the nix packages website)

### Proposal

Could you maybe create a nix package containing volare?

**Comment by RK4444** [comment]
Thank you very much. I didn't know there was a package since I haven't found it on the nix packages search page. Thank you very much again.

Edit: It works when I import it as a package, but it complains that I wouldn't have an internet connection which is weird because I downloaded it just a moment ago

**Comment by donn** [comment]
Could you invoke `nix run github:efabless/volare -- ls-remote 2> ls-remote-error.log` and upload the `ls-remote-error.log` file please?

**Comment by RK4444** [comment]
When I execute with the command you stated, everything works fine. However if I try to import volare into my own nix flake, ls-remote fails. Here is the error log:
[ls-remote-error.log](https://github.com/user-attachments/files/15826170/ls-remote-error.log)

When I import it, I first add   ``github:efabless/volare`` to my inputs, then I extract the package like ``volare = efablesspkgs.packages.${system}.default;`` and finally add the package to the ``nativeBuildInputs``. Is this right this way?

**Comment by donn** [comment]
That's not a Volare issue, then- it's a Nix thing. I can still answer though:

Nix only gives network (and thus, internet) access to what are known as "fixed-output derivations", i.e., derivations for which the output already has a pre-computed hash that is verified upon re-build. See this blog post for more: https://bmcgee.ie/posts/2023/02/nix-what-are-fixed-output-derivations-and-why-use-them/

With respect to making a fixed-output-derivation out of Volare: we currently make no guarantees that results of executing `volare fetch` or `volare enable` will always be the same, even for the same version: we have done updates and re-uploaded files to fix issues.

Long story short: there's a reason why we haven't moved the PDK to Nix as well. But you're welcome to try! 🙂

**Comment by RK4444** [comment]
Thank you very much for the answer. I kind of expected something like this but due to the fact that it seems to work with openlane 2 I thought I could achieve something similar. Since I use nix with wsl I suspected windows of blocking something (also because I can't install any package with pip in the flake) but according to your explanation that doesn't seem the case. I will try a bit further, if you are interested I could notify you in this issue when I achieve it. I only have 3 weeks of nix experience until now, so I don't think I can do that, but we'll see. Thank you anyway for your work. I really appreciate it.
